src/containers/Noticeboard.jsx

```diff
@@ -1,125 +1,94 @@
+    this.props.fetchNotices(this.state.noticeType);
+  };
 
-  _handleWaypointLeave() {
-  }
+  _handleWaypointLeave = () => {};
 
   render() {
+    const { notices, isFetching } = this.props;
+    const { noticeType } = this.state;
+    const { data } = notices[noticeType];
     return (
       <div className="section-main">
         <div className="section-title">Noticeboard</div>
         <div className="type-list">
           <ul>
-            <li>
-              <a className={this.state.notice_type === 0? 'active type-button' : 'type-button'} onClick={() => this.switchType(0)}>UG</a>
-            </li>
-            <li>
-              <a className={this.state.notice_type === 1? 'active type-button' : 'type-button'} onClick={() => this.switchType(1)}>PG</a>
-            </li>
-            <li>
-              <a className={this.state.notice_type === 2? 'active type-button' : 'type-button'} onClick={() => this.switchType(2)}>BC Roy</a>
-            </li>
-            <li>
-              <a className={this.state.notice_type === 3? 'active type-button' : 'type-button'} onClick={() => this.switchType(3)}>General</a>
-            </li>
+            {["UG", "PG", "BC Roy", "General"].map((str, idx) => (
+              <li key={idx}>
+                <a
+                  className={
+                    noticeType === idx ? "active type-button" : "type-button"
+                  }
+                  onClick={() => this.switchType(idx)}
+                >
+                  {str}
+                </a>
+              </li>
+            ))}
           </ul>
         </div>
         <div className="notice-list">
-          {
-            this.state.notices.map((notice, i) =>
-              <Noticecard
-                key={i}
-                title={notice.title}
-                time={notice.time}
-                content={notice.html}
-                attachment={notice.attachment}
-              />
-            )
-          }
+          {data.map((notice, i) => (
+            <Noticecard
+              key={i}
+              // title={notice.title}
+              // time={notice.time}
+              // content={notice.html}
+              // attachment={notice.attachment}
+              {...notice}
+            />
+          ))}
           <Waypoint
             onEnter={this._handleWaypointEnter}
             onLeave={this._handleWaypointLeave}
           />
-          {(() => {
-            if (!this.state.all_fetched) {
-              return (
-                <Code
-                  className='placeholderNotice'
-                  height={50}
-                  primaryColor='#fff'
-                />
-              );
-            }
-          })()}
+          {isFetching ? (
+            <Code
+              className="placeholderNotice"
+              height={50}
+              primaryColor="#fff"
+            />
+          ) : (
+            ""
+          )}
         </div>
       </div>
-    )
+    );
   }
 }
 
+Noticeboard = connect(
```

Comment:
This is a bit confusing. Noticeboard is also the name of the class and this line is assigning to that. Would it make more sense to just do this instead?
```js
export default connect(....)(Noticeboard);
```

Reply:
Yes, you're right. If decorators needed to be chained then this syntax would make more sense.